src/averageAges.js

```diff
@@ -56,6 +74,21 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+
+  const mothers = people.filter(person => person.sex === 'f'
+    && people.some(women => women.mother === person.name));
+
+  const children = people.filter(person => onlyWithSon
+    ? mothers.some(mother => mother.name === person.mother)
+    && person.sex === 'm'
+    : mothers.some(mother => mother.name === person.mother));
+
+  const ageDifference = children.reduce((totalAgeDiff, child) =>
+    totalAgeDiff + (child.born - (mothers.find(mother => mother.name
+    === child.mother)).born), 0
+  ) / children.length;
```

Comment:
```suggestion
  const ageDifference = children.reduce(
    (totalAgeDiff, child) => (
      totalAgeDiff + (child.born - (mothers.find(mother => (
        mother.name=== child.mother
      ))).born)
    ), 
    0,
  ) / children.length;
```

Reply:
if I would writing this, I'd follow some rules
- if the parameter of the function very long, I would use a new line for every parameter
- if I want to return immediately something from the arrow function I open brace `(` and start from the new line (don't forget to close brace `)`)

Reply:
also would be nice to save somewhere the result of the find method because, if you got `undefined` you can't use it in math operations, if you save it somewhere you can check if the result is a number before using it

Reply:
and try to avoid so many nested methods, it's hard to read and very hard to debug